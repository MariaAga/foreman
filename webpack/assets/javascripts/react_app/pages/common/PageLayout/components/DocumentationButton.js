import React from 'react';
import { Button, Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from '../../../../common/I18n';

const DocumentationButton = ({ url, text }) => (
  <Button href={url} className="btn-docs">
    <Icon type="pf" name="help" />
    {` ${text}`}
  </Button>
);

DocumentationButton.propTypes = {
  url: PropTypes.string.isRequired,
  text: PropTypes.string,
};

DocumentationButton.defaultProps = {
  text: __('Documentation'),
};

export default DocumentationButton;
