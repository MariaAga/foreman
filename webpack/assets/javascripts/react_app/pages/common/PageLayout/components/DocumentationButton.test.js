import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import DocumentationButton from './DocumentationButton';

const fixtures = {
  'render with minimal props': { url: 'url' },
};

describe('DocumentationButton', () =>
  testComponentSnapshotsWithFixtures(DocumentationButton, fixtures));
