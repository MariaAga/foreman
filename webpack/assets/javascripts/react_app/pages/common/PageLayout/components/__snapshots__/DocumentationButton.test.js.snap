// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentationButton render with minimal props 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="btn-docs"
  disabled={false}
  href="url"
>
  <Icon
    name="help"
    type="pf"
  />
   Documentation
</Button>
`;
